syntax = "proto3";

package silenteight.agent.bank_identification_codes_agent.v1.api;

option java_package = "com.silenteight.proto.agent.bank_identification_codes_agent.v1.api";
option java_outer_classname = "BankIdentificationCodesAgentApiProto";
option java_multiple_files = true;

service BankIdentificationCodesAgent {

  rpc CheckBankIdentificationCodes (CheckBankIdentificationCodesRequest)
      returns (CheckBankIdentificationCodesResponse) {
  }
}

message CheckBankIdentificationCodesRequest {
  string altered_party_matching_field = 1;
  string watchlist_matching_text = 2;
  string watchlist_type = 3;
  repeated string watchlist_search_codes = 4;
  repeated string watchlist_bic_codes = 5;
}

message CheckBankIdentificationCodesResponse {
  string solution = 1;
  CheckBankIdentificationCodesReason reason = 2;
}

message CheckBankIdentificationCodesReason {
    string conclusion = 1;
    string altered_party_matching_sequence = 2;
    string altered_party_matching_field = 3;
    string partial_match_text = 4;
    string watchlist_matching_text = 5;
    string watchlist_type = 6;
    repeated string watchlist_search_codes = 7;
    string watchlist_bic_code = 8;
}
